temporian/implementation/numpy/operators/drop_index.py

```diff
@@ -0,0 +1,161 @@
+from collections import defaultdict
+from typing import Dict, List, Union
+
+import numpy as np
+
+from temporian.core.operators.drop_index import DropIndexOperator
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+from temporian.implementation.numpy.data.sampling import NumpySampling
+
+IndexMetadata = Dict[
+    str, Union[int, List[List[NumpyFeature]], List[np.ndarray]]
+]
+
+
+class DropIndexNumpyImplementation:
+    def __init__(self, operator: DropIndexOperator) -> None:
+        """
+        Initializes an instance of the DropIndexNumpyImplementation class.
+
+        Args:
+            operator: The DropIndexOperator instance.
+        """
+        self.operator = operator
+
+    def __call__(self, event: NumpyEvent) -> Dict[str, NumpyEvent]:
+        """
+        Calls the implementation function with the specified event.
+
+        Args:
+            event: The input NumpyEvent.
+
+        Returns:
+            A dictionary with the modified NumpyEvent.
+        """
+        # get attributes from operator
+        drop_index_names = self.operator.index_names
+        keep = self.operator.keep
+        dst_index_names = self.operator.dst_index_names
+        dst_feat_names = self.operator.dst_feat_names
+
+        # source index names
+        src_index_names = event.sampling.index
+
+        # destination feature dtypes
+        dst_dtypes = (
+            {**event.sampling.dtypes, **event.dtypes} if keep else event.dtypes
+        )
+        # index positions to keep
+        keep_index_pos = [
+            pos
+            for pos, idx_name in enumerate(src_index_names)
+            if idx_name in dst_index_names
+        ]
+        # index positions to drop
+        drop_index_pos = [
+            pos
+            for pos, idx_name in enumerate(src_index_names)
+            if idx_name not in dst_index_names
+        ]
+        # intialize empty dict mapping destination index levels to block lengths,
+        # features, and timestamps
+        dst_index_metadata: Dict[tuple, IndexMetadata] = defaultdict(
+            lambda: {"block_length": 0, "timestamps": [], "features": []}
```

Comment:
Similarly as for the DropIndex implementation. I would use a class or named tuple instead.